Documentation/ApiOverview/Bootstrapping/Index.rst

```diff
@@ -7,116 +7,112 @@ Bootstrapping
 =============
 
 TYPO3 CMS has a clean bootstrapping process driven mostly
-by class :php:`\TYPO3\CMS\Core\Core\Bootstrap`. This class
-contains a host of methods each responsible for a little
-step along the initialization of a full TYPO3 process,
-be it the backend or other contexts.
+by class :php:`\TYPO3\CMS\Core\Core\Bootstrap`. This class is initialized by
+calling  :php:`Bootstrap::init()` and serves as an entry point for later calling
+an Application class, depending on the TYPO3_MODE and several context-dependant
+constraints.
 
-Some contexts add their own bootstrap class (like the command
-line, which additionally requires :php:`\TYPO3\CMS\Core\Core\CliBootstrap`).
+Each Application class registers Request Handlers to the TYPO3 Bootstrap to
+run a certain request type (e.g. eID or TSFE-logic, or AJAX requests in the Backend). Each Application is handed
+over the Class Loader provided by Composer.
 
-.. note::
-
-   The frontend's bootstrapping process is not yet fully encapsulated
-   in a bootstrap class.
+Applications
+""""""""""""
 
-.. warning::
+There are four types of Applications provided by the TYPO3 Core:
 
-   This bootstrapping API is internal and may change at any time in the near future
-   even in minor updates. It is thus discouraged to use it in third party code.
-   Use this class only if other extensibility possibilities such as
-   :ref:`Hooks <hooks>`, Signals or :ref:`XCLASS <xclasses>`
-   are not enough to reach your goals.
+\\TYPO3\\CMS\\Frontend\\Http\\Application
+-----------------------------------------
+All incoming web requests coming to index.php in the main directory, handling all TSFE and eID requests.
+The Application sets TYPO3_MODE=FE very early.
+The Application checks if all configuration is given, otherwise redirects to the TYPO3 Install Tool.
 
+\\TYPO3\\CMS\\Backend\\Http\\Application
+----------------------------------------
+All incoming web requests for any regular Backend call inside typo3/\*. This handles three types of Request Handlers:
```

Comment:
e.g. 

> This class handles all incoming web request ... Three types of request handlers are registered:

Reply:
there is actually just one in the current master